script.js

```diff
@@ -1,12 +1,187 @@
+  searchResult.innerHTML = `Displaying ${arr.length}/${arr.length}`;
+  searchShowsEl.setAttribute('style', 'display: none !important');
+  searchEpisodesEl.setAttribute('style', 'display: inline');
+}
+
+function createEpisodeSelect(arr) {
+  selectEpisodesEl.innerHTML = '';
+  let selectAllEpisodes = document.createElement('option');
+  selectAllEpisodes.innerText = 'All Episodes';
+  selectAllEpisodes.value = 'selectAll';
+  selectEpisodesEl.appendChild(selectAllEpisodes);
+  arr.forEach(episode => {
+    let selectOption = document.createElement('option');
+    selectOption.innerText = `${getEpisodeTitle(episode)}`;
+    selectOption.value = episode.id;
+    selectEpisodesEl.appendChild(selectOption);
+  });
+}
+
+function displaySelectedShow(arr) {
+  createEpisodeSelect(arr);
+  displayEpisodes(arr);
+  searchEpisodesEl.value = '';
+  searchShowsEl.setAttribute('style', 'display: none !important');
+  searchEpisodesEl.setAttribute('style', 'display: inline');
+}
+
+selectEpisodesEl.addEventListener('change', (e) => {
+  if (e.target.value == 'selectAll') { displayEpisodes(allEpisodes); }
+  for (let ep of allEpisodes) {
+    if (ep.id == selectEpisodesEl.value) {
+      episodesEl.innerHTML = getEpisodeCard(ep);
+      searchResult.innerHTML = `Displaying 1/${allEpisodes.length}`;
+    }
+  }
+})
+
+selectShowEl.addEventListener('change', (e) => {
+  if (e.target.value == 'allShows') {
+    displayShows(allTVShows);
+    searchEpisodesEl.value = '';
+  };
+  for (let show of allTVShows) {
+    if (show.id == selectShowEl.value) { fetchShowAndDisplay(show.id); }
+  }
+})
+
+function createShowSelect(arr) {
+  selectShowEl.innerHTML = '';
+  let selectAllShows = document.createElement('option');
+  selectAllShows.innerText = 'All Shows';
+  selectAllShows.value = 'allShows';
+  selectShowEl.appendChild(selectAllShows);
+  arr.forEach(show => {
+    let selectOption = document.createElement('option');
+    selectOption.innerText = show.name;
+    selectOption.value = show.id;
+    selectShowEl.appendChild(selectOption);
+  })
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+function displayShows(arr) {
+  showsEl.setAttribute('style', 'display:inline');
+  episodesEl.setAttribute('style', 'display:none !important');
+  let allShowsHTML = '';
+  arr.forEach(show => allShowsHTML += getShowCard(show))
+  showsEl.innerHTML = allShowsHTML;
+  searchResult.innerHTML = `Displaying ${arr.length}/${arr.length}`;
+  searchShowsEl.setAttribute('style', 'display: inline');
+  searchEpisodesEl.setAttribute('style', 'display: none !important');
 }
 
-window.onload = setup;
+function displayAllShows(arr) {
+  createShowSelect(arr);
+  displayShows(arr);
+  searchShowsEl.value = '';
+  searchShowsEl.setAttribute('style', 'display: inline');
+  searchEpisodesEl.setAttribute('style', 'display: none !important');
```

Comment:
Be careful about your use of `!important`. Can be hard to override later due to its specificity.